packages/metapackage/test/completer/handler.spec.ts

```diff
@@ -317,6 +346,88 @@ describe('@jupyterlab/completer', () => {
       });
     });
 
+    describe('#CompletionTriggerKind', () => {
+      it('should use CompletionTriggerKind.TriggerCharacter', () => {
+        const provider = new FooCompletionProvider(true);
+        const reconciliator = new ProviderReconciliator({
+          context: null as any,
+          providers: [provider],
+          timeout: 0
+        });
+
+        const handler = new CompletionHandler({
+          reconciliator,
+          completer: new Completer({ editor: null })
+        });
+        handler.editor = createEditorWidget().editor;
+
+        expect(provider.methods.length).toEqual(0);
+
+        handler.editor.model.sharedModel.setSource('foo.');
+        // @ts-ignore
+        handler._makeRequest(
+          { line: 0, column: 3 },
+          CompletionTriggerKind.TriggerCharacter
+        );
+
+        expect(provider.methods).toEqual(
+          expect.arrayContaining([CompletionTriggerKind.TriggerCharacter])
+        );
+      });
+
+      it('should use CompletionTriggerKind.Invoked', () => {
+        const provider = new FooCompletionProvider(true);
+        const reconciliator = new ProviderReconciliator({
+          context: null as any,
+          providers: [provider],
+          timeout: 0
+        });
+
+        const handler = new CompletionHandler({
+          reconciliator,
+          completer: new Completer({ editor: null })
+        });
+        handler.editor = createEditorWidget().editor;
+
+        expect(provider.methods.length).toEqual(0);
+
+        handler.editor.model.sharedModel.setSource('foo.');
+        // @ts-ignore
+        handler._makeRequest(
+          { line: 0, column: 3 },
+          CompletionTriggerKind.Invoked
+        );
```

Comment:
It ain't end-to-end but I guess good enough for now.

Reply:
@krassowski, I improved the tests in my last commit. please take a look at them and let me know.